kikuchipy/__init__.py

```diff
@@ -20,26 +20,41 @@
 from kikuchipy.release import version as __version__
 
 
-# Attempt import only once
+# Attempt (and fail) import of optional dependencies only once
 try:
     import pyvista
 
     _pyvista_installed = True
 except ImportError:  # pragma: no cover
     _pyvista_installed = False
+
 try:
     import nlopt
 
     _nlopt_installed = True
 except ImportError:  # pragma: no cover
     _nlopt_installed = False
+
 try:
     from pyebsdindex import pcopt, ebsd_index
 
     _pyebsdindex_installed = True
 except ImportError:  # pragma: no cover
     _pyebsdindex_installed = False
 
+# PyOpenCL context available for use with PyEBSDIndex?
+try:
+    from pyebsdindex.opencl.openclparam import OpenClParam
+
+    clparams = OpenClParam()
+    clparams.get_context()
+    assert clparams.ctx is not None  # pragma: no cover
```

Comment:
Off the top of my head: using `assert` is intended for development and testing and is not a good pattern to catch runtime errors.

Reply:
Thanks, I've replaced it with an if/else.